tests/TestGUI.py

```diff
@@ -83,122 +109,124 @@ def confirm_current_test(test_data_path: str) -> bool:
                 TestGUI.tests[test]['button'].config(state="disabled")
                 TestGUI.tests[test]['button'].config(bg="green")
 
-        # Clear graphs from memory
-        plt.close()
+        # Reset the timer
+        TestGUI.reset_timer()
 
         # Return true if test is complete
-        return TestGUI.tests[TestGUI.current_test]['completed']
+        return TestGUI.tests[TestGUI.current_thread.name]['completed']
 
     @staticmethod
-    def start_test(test_name):
+    def start_test(test_thread):
         """
         A function to disable buttons while a test is running.
 
-        :param test_name: The name of the current test running (to enable the indicator).
+        :param test_thread: The thread object of the current test running (to enable the indicator).
         """
         # Disable all buttons
         TestGUI.close_button.config(state="disabled")
+        TestGUI.abort_button.config(state="normal")
 
         for test in TestGUI.tests.keys():
             TestGUI.tests[test]['button'].config(state="disabled")
 
         # Indicate current test
-        TestGUI.current_test = test_name
-        TestGUI.tests[TestGUI.current_test]['button'].config(bg="yellow")
+        TestGUI.current_thread = test_thread
+        TestGUI.tests[TestGUI.current_thread.name]['button'].config(bg="yellow")
+
+        # Initialize test start time
+        TestGUI.test_start_time = time.time()  # Record the start time
+
+        # Start the timer
+        TestGUI.timer_thread = threading.Thread(target=TestGUI.update_timer)
+        TestGUI.timer_thread.start()
+
+        TestGUI.test_start_time = time.time()  # Record the start time
 
     #
     # HELPER METHODS
     #
 
     @staticmethod
-    def __show_data_and_confirm(test_data_path: str):
+    def update_timer():
```

Comment:
If it is a "helper function" only used within the file/class and should be private it should start with two underscores like this:
```
def __update_timer():
```

Reply:
I meant single underscores